internal/auth.go

```diff
@@ -93,9 +94,14 @@ func redirectBase(r *http.Request) string {
 
 // Return url
 func returnUrl(r *http.Request) string {
-	path := r.Header.Get("X-Forwarded-Uri")
-
-	return fmt.Sprintf("%s%s", redirectBase(r), path)
+	replacedPath := r.Header.Get("X-Replaced-Path")
+	if replacedPath != "" {
+		return fmt.Sprintf("%s%s", redirectBase(r), replacedPath)
+	}
+	uri := r.Header.Get("X-Forwarded-Uri")
+	prefix := r.Header.Get(("X-Forwarded-Prefix"))
+	p := path.Join(prefix, uri)
+	return fmt.Sprintf("%s%s", redirectBase(r), p)
```

Comment:
I think we should extract the path parsing logic into a separate function that we use here and in the server `RootHandler`?

Reply:
I don't think a wrapper around `fmt.Sprintf` is usefull, if that's what you mean...